gdpr/impl_test.go

```diff
@@ -79,6 +80,9 @@ func TestAllowedSyncs(t *testing.T) {
 			}),
 		},
 	}
+	perms.purposeConfigs = map[consentconstants.Purpose]config.TCF2Purpose{
+		consentconstants.Purpose(1): perms.cfg.TCF2.Purpose1,
+	}
```

Comment:
`vendor2AndPurpose1Consent` was defined above, should we add `consentconstants.Purpose(2)` to the map instead of `consentconstants.Purpose(1)`?

`TestProhibitedVendors` defines `purpose1NoVendorConsent` which is passed as third parameter to `perms.HostCookiesAllowed` and `perms.BidderSyncAllowed` and includes `consentconstants.Purpose(1)` in the `perms.purposeConfigs` map, should we expect the same in this test?

Reply:
This is ok as is. In this test and `TestProhibitedVendors`, we are using vendor 2 and purpose 1 to exercise different code paths.

In this test, we are verifying that PBS determines host cookies and syncs are allowed when the config says to enforce vendors, which means that the consent string must indicate the user consents to the purpose and to the vendor (which is what the `vendor2AndPurpose1Consent` variable name indicates).

In the `TestProhibitedVendors` test, everything is the same except that the user consents to the purpose but not the vendor (which is what the `purpose1NoVendorConsent` variable name is meant to indicate) so the host cookies allowed and sync allowed checks return false instead of true.